porch/pkg/git/git.go

```diff
@@ -286,20 +286,39 @@ func (r *gitRepository) CreatePackageRevision(ctx context.Context, obj *v1alpha1
 		return nil, fmt.Errorf("error when resolving target branch for the package: %w", err)
 	}
 
-	draft := createDraftName(obj.Spec.PackageName, obj.Spec.Revision)
+	if obj.Spec.Description == "" {
```

Comment:
We should definitely add this validation in the api layer as well, as well as here. Checks that only affects the current resource (old and new) should probably be in the validator. Checks for uniqueness is a bit more challenging. Do we currently do any checks to verify uniqueness of revisions? I seem to remember that we have a bug for this. It seems like guaranteeing uniqueness will be difficult since users can also create packagerevisions directly in git/oci.

Reply:
We could also check uniqueness against the branches stored in git/oci ... that's what matters I believe.

git has the ability to push a branch only if it does not exist, oci does not (yet).  We need to do some work to plumb through the "before" state so we can do a compare-and-swap operations against git, but I don't think we want to tackle that in this PR!

Reply:
Yeah, I agree that for git we should leverage uniqueness of branches to handle this. Also agree that we should handle it separately.

Reply:
Added the empty description validation in the api layer. I agree that leveraging git branches would be helpful for uniqueness, but I can tackle that in a future PR if we should handle that separately.